otelcollector/configmapparser/tomlparser-default-scrape-settings.rb

```diff
@@ -93,27 +98,54 @@ def populateSettingValuesFromConfigMap(parsedConfig)
       @kappiebasicEnabled = parsedConfig[:kappiebasic]
       puts "config::Using configmap scrape settings for kappiebasic: #{@kappiebasicEnabled}"
     end
+    if !parsedConfig[:"kube-controller-manager"].nil?
+      @kubecontrollermanagerEnabled = parsedConfig[:"kube-controller-manager"]
+      puts "config::Using configmap scrape settings for kube-controller-manager: #{@kubecontrollermanagerEnabled}"
+    end
+    if !parsedConfig[:"kube-scheduler"].nil?
+      @kubeschedulerEnabled = parsedConfig[:"kube-scheduler"]
+      puts "config::Using configmap scrape settings for kube-scheduler: #{@kubeschedulerEnabled}"
+    end
+    if !parsedConfig[:"kube-apiserver"].nil?
+      @kubeapiserverEnabled = parsedConfig[:"kube-apiserver"]
+      puts "config::Using configmap scrape settings for kube-apiserver: #{@kubeapiserverEnabled}"
+    end
+    if !parsedConfig[:"cluster-autoscaler"].nil?
+      @clusterautoscalerEnabled = parsedConfig[:"cluster-autoscaler"]
+      puts "config::Using configmap scrape settings for cluster-autoscaler: #{@clusterautoscalerEnabled}"
+    end
+    if !parsedConfig[:"etcd"].nil?
+      @etcdEnabled = parsedConfig[:"etcd"]
+      puts "config::Using configmap scrape settings for etcd: #{@etcdEnabled}"
+    end
 
     windowsDaemonset = false
     if ENV["WINMODE"].nil? && ENV["WINMODE"].strip.downcase == "advanced"
       windowsDaemonset = true
     end
 
+    # ccp-metrics addon settings for api-server (old flag) and kube-apiserver (new flag)
+    if apiserverEnabled && kubeapiserverEnabled
+      # honor the old flag, this is a very unlikely scenario
+      kubeapiserverEnabled = false
+    end
+
+    ccpmetricsEnabled = kubecontrollermanagerEnabled || kubeschedulerEnabled || kubeapiserverEnabled || clusterautoscalerEnabled || etcdEnabled
```

Comment:
do these do be referenced with @ prefix? Looks like it based on how they are declared?